### PR TITLE
Traditional headermap / manual order annoyance fix

### PR DESCRIPTION
**Pull request by dimitre**
Useful? no. informative? no. annoying? yes
 
Screen Shot 2022-04-12 at 23 52 56

### CODE REVIEW

**Comment by ofTheo** [comment]
@dimitre I had been nervous about this one as I thought it meant we could no longer do `#include "ofXXX.h"` but I think the fix is simply to set `ALWAYS_SEARCH_USER_PATHS` to no. In both the OF project and the project template. 

For me it stops the warning and everything still compiles normally. 

I think even removing the plist entry in the two files could be enough as the default is now `NO` anyway. 

Thanks!
Theo

**Comment by dimitre** [comment]
Great. In fact I've just tried to remove and the issue comes back I suppose the default is still YES. 
It is working OK here in different projects. 
I didn't understand the #include "ofXXX.h" issue. can you send a quick example or just test if this actually happen?
Thanks!

**Comment by ofTheo** [comment]
I was worried that we would have to do stuff like `#include "openframeworks/utils/ofFileUtils.h"` instead of  `#include "ofFileUtils.h"` but when I tried setting both to NO and rebuilding OF and an example project it all compiled fine.  So I think this could be a safer change to make. 🙂

File: scripts/templates/osx/Project.xcconfig [hunk 1]
```diff
@@ -22,6 +22,8 @@ OTHER_LDFLAGS = $(OF_CORE_LIBS) $(OF_CORE_FRAMEWORKS)
 HEADER_SEARCH_PATHS = $(OF_CORE_HEADERS)
 
 //COMPILER SETTINGS WHICH CAN BE OVERRIDDEN BY XCODE BUILD SETTINGS
+ALWAYS_SEARCH_USER_PATHS = NO
```
[the remark this answers is not in the document]
Comment (dimitre) [reply scripts/templates/osx/Project.xcconfig:25]: good catch!
I've added in this other PR now which is already working with shared xcconfig. 
https://github.com/openframeworks/openFrameworks/pull/6941

[the remark this answers is not in the document]
Comment (ofTheo) [reply scripts/templates/osx/Project.xcconfig:25]: cool - I'll close this one. 
does adding it there it remove those warning messages?

[the remark this answers is not in the document]
Comment (dimitre) [reply scripts/templates/osx/Project.xcconfig:25]: haha sorry it was both. this one for the project template and the other one for OF

**Comment by ofTheo** [comment]
@dimitre looking at the xcconfig structure anything that should be set both for the OF lib and the Project can be set in `CoreOF.xcconfig`  

Putting `ALWAYS_SEARCH_USER_PATHS = NO` in `CoreOF.xcconfig` fixes the warning for both the OF build and the Project build.   So nice that we don't have to have it duplicated in multiple places. 

So in short: 
- settings that should be shared across all builds ( Project Debug, Project Release, OF Lib etc ) -> `CoreOF.xcconfig`  
- settings that should be just for OF Lib -> `Shared.xcconfig`  
- settings that should be just for OF Lib Debug or Release -> `Debug.xcconfig`  or `Release.xcconfig`  
- settings for just the user's Project -> `Project.xcconfig` 

Sorry for not clarifying sooner, I actually thought `Shared.xcconfig` was the config that everything pulled in from. 🤦‍♂️

**Comment by dimitre** [comment]
Great @ofTheo thanks for the explanation.
I've moved the configurations from Shared to CoreOF in this PR
https://github.com/openframeworks/openFrameworks/pull/6941

And this one here can be closed.

**Pull request closed without merging**